crates/pipeline-manager/src/api/endpoints/pipeline_management.rs

```diff
@@ -670,12 +710,116 @@ pub(crate) async fn list_pipelines(
+                .lock()
+                .await
+                .list_pipelines_for_monitoring(*tenant_id)
+                .await?;
+
+            // Fetch connector stats for all pipelines in parallel
+            let stats_futures: Vec<_> =
+                pipelines
+                    .iter()
+                    .map(|pipeline| {
+                        let state = state.clone();
+                        let tenant_id = *tenant_id;
+                        let pipeline_name = pipeline.name.clone();
+                        async move {
+                            fetch_connector_error_stats(&state, tenant_id, &pipeline_name).await
+                        }
+                    })
+                    .collect();
+
+            let connector_stats = join_all(stats_futures).await;
+
+            pipelines
+                .into_iter()
+                .zip(connector_stats.into_iter())
+                .map(|(pipeline, stats)| {
+                    PipelineSelectedInfoInternal::new_status_with_connectors(pipeline, stats)
+                })
+                .collect()
+        }
     };
     Ok(HttpResponse::Ok()
         .insert_header(CacheControl(vec![CacheDirective::NoCache]))
         .json(returned_pipelines))
 }
 
+/// Fetch and aggregate connector error statistics for a pipeline.
+///
+/// Returns `None` if the pipeline is unavailable or if there's an error fetching stats.
+async fn fetch_connector_error_stats(
+    state: &WebData<ServerState>,
+    tenant_id: TenantId,
+    pipeline_name: &str,
+) -> Option<ConnectorStats> {
+    // Use the existing method to forward the request to the pipeline
+    let client = awc::Client::default();
+    let response = state
+        .runner
+        .forward_http_request_to_pipeline_by_name(
+            &client,
+            tenant_id,
+            pipeline_name,
+            actix_web::http::Method::GET,
+            "stats",
+            "",
+            Some(std::time::Duration::from_millis(500)),
+        )
+        .await
+        .ok()?;
+
+    // Parse the response body
+    let body = response.into_body();
+    let bytes = actix_web::body::to_bytes(body).await.ok()?;
+    let stats_response: serde_json::Value = serde_json::from_slice(&bytes).ok()?;
+
+    // Extract and aggregate error counts
+    let mut total_errors = 0u64;
+
+    // Aggregate input connector errors
+    if let Some(input_endpoints) = stats_response.get("inputs").and_then(|v| v.as_object()) {
+        for endpoint in input_endpoints.values() {
+            if let Some(metrics) = endpoint.get("metrics").and_then(|v| v.as_object()) {
+                if let Some(transport_errors) =
+                    metrics.get("num_transport_errors").and_then(|v| v.as_u64())
+                {
+                    total_errors = total_errors.saturating_add(transport_errors);
+                }
+                if let Some(parse_errors) = metrics.get("num_parse_errors").and_then(|v| v.as_u64())
+                {
```

Comment:
While this works, it might be better to just create a type, inside the function with only the fields we are interested in, and ask serde to parse the JSON.